File: Regular/925. Long Pressed Name/Solution.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;
class Solution {
    public boolean isLongPressedName(String name, String typed) {
        String p = "";
        int count = 1;
        int i;
        for (i = 1; i < name.length(); i++) {
            if (name.charAt(i - 1) == name.charAt(i)) count ++;
            else {
                p += name.charAt(i - 1) + "{" + count + ",}";
                count = 1;
            }
        }
        p += name.charAt(i - 1) + "{" + count + ",}";
        return Pattern.compile(p).matcher(typed).matches();
    }
}

File: Regular/925. Long Pressed Name/Solution.cpp
class Solution {
public:
    bool isLongPressedName(string name, string typed) {
        if (name == typed) return true;
        char cLast = 0;
        for (char c : name) {
            if (typed.length() == 0) return false;
            if (typed.at(0) == c) {
                typed.erase(0, 1);
                cLast = c;
            }
            else if (typed.at(0) == cLast) {
                while (typed.length() > 0 && typed.at(0) == cLast) typed.erase(0, 1);
                if (typed.length() > 0 && typed.at(0) == c) {
                    typed.erase(0, 1);
                    cLast = c;
                }
                else return false;
            }
            else return false;
        }
        return true;
    }
};
